### DISCUSSION TITLE
how ı create popup window

### DISCUSSION THREAD

**Discussion by ghost** [qtile/qtile#2866] [Q&A] [answered]
from tkinter import Tk

window = Tk()

window.title("test")
window.geometry("300x200")
window.mainloop()

yes ı learn python but not working just create tiling window

**Comment by m-col** [comment, marked as answer]
Hi @ArchKubi, is the problem that you want the window to be floating and not tiled?

**Reply by ghost** [reply]
normal popup window 
ı try create application finder basic version
for example
![2021-10-09-155929_1280x1012_scrot](https://user-images.githubusercontent.com/80429360/136658864-fe142d07-3406-405c-9bd5-7268179ac6cd.png)

**Reply by m-col** [reply]
You need to add the title of your window to your `float_rules` in your config. E.g. if you are using `window.title("test")`, then you need to add `Match(title='title')` to your `float_rules`.

**Reply by ghost** [reply]
ı add this float rules config.py ? ı don't understand ı'm noob

**Reply by m-col** [reply]
> ı add this float rules config.py ?

Yes

**Reply by ghost** [reply]
float rules = True ? or please give some example

**Reply by m-col** [reply]
Please see the default config for example usage.

**Reply by ghost** [reply]
ı edit default config.py 
![2021-10-09-161546_540x192_scrot](https://user-images.githubusercontent.com/80429360/136659305-b4152aab-5d64-44ab-98a0-cd040becba61.png)

**Reply by elParaguayo** [reply]
Hi @ArchKubi, it's important to understand that Qtile is a tiling window manager. This means that the manager automatically fits windows to the screen depending on the selected layout. So, when you create a window, Qtile will try to position that window based on the current layout. You need to tell Qtile that you don't want it to tile the window and you do this by telling Qtile to "float" it.

If you're new to tiling window managers, Qtile and the config file then it might be worth having a read of the [docs](http://docs.qtile.org/en/latest/) to help familiarise yourself.

We're happy to help when you have questions too.

**Reply by elParaguayo** [reply]
> You need to add the title of your window to your `float_rules` in your config. E.g. if you are using `window.title("test")`, then you need to add `Match(title='title')` to your `float_rules`.

As m-col says, you need to add it to your float_rules.

**Reply by ghost** [reply]
okay ı understand now

**Reply by elParaguayo** [reply]
So, `float_rules` is a list (it's enclosed in `[` and `]`) with each item separated by a comma. For `float_rules`, you can see there are 7 items in the list. The first is `*layout.Floating.default_float_rules` and then there are 6 `Match` items. 

To add your item to the list, you need to just add it after the last `Match` item and before the `]`.

**Reply by ghost** [reply]
thanks bro
![2021-10-09-162631_1280x1012_scrot](https://user-images.githubusercontent.com/80429360/136659754-734ca963-d333-4571-9659-9f4c556c2f68.png)

**Reply by elParaguayo** [reply]
Nice work!
